core/src/avm2/globals.rs

```diff
@@ -498,6 +501,10 @@ pub fn load_player_globals<'gc>(
     );
     avm2_system_class!(array, activation, array::create_class(activation), script);
 
+    // TODO: this should _not_ be exposed as a ClassObject, getDefinitionByName etc.
+    // it should only be visible as an type for typecheck/cast purposes.
```

Comment:
It might be worth pointing out the additional limitation that this errors if used for a purpose other than as a return type (you cannot coerce to void using the coerce opcode, for example)

Reply:
For the record, it _can_ be used for the `coerce` opcode (it's not much useful ofc - but it throws a runtime exception instead of an early VerifyError, so it's a perfectly normal coercion failure).